firebase_admin/_user_mgt.py

```diff
@@ -331,6 +378,97 @@ def iterate_all(self):
         return _UserIterator(self)
 
 
+class DeleteUsersResult:
+    """Represents the result of the ``auth.delete_users()`` API."""
+
+    def __init__(self, result, total):
+        """Constructs a DeleteUsersResult.
+
+        Args:
+          result: BatchDeleteAccountsResponse: The proto response, wrapped in a
+              BatchDeleteAccountsResponse instance.
+          total: integer: Total number of deletion attempts.
+        """
+        errors = result.errors
+        self._success_count = total - len(errors)
+        self._failure_count = len(errors)
+        self._errors = errors
+
+    @property
+    def success_count(self):
+        """Returns the number of users that were deleted successfully (possibly
+        zero).
+
+        Users that did not exist prior to calling delete_users() will be
+        considered to be successfully deleted.
+        """
+        return self._success_count
+
+    @property
+    def failure_count(self):
+        """Returns the number of users that failed to be deleted (possibly
+        zero).
+        """
+        return self._failure_count
+
+    @property
+    def errors(self):
+        """A list of ``auth.BatchDeleteErrorInfo`` instances describing the
+        errors that were encountered during the deletion. Length of this list
+        is equal to `failure_count`.
+        """
+        return self._errors
+
+
+class BatchDeleteErrorInfo:
+    """Represents an error that occurred while attempting to delete a batch of
+    users.
+    """
+
+    def __init__(self, err):
+        """Constructs a BatchDeleteErrorInfo instance, corresponding to the
+        json representing the BatchDeleteErrorInfo proto.
+
+        Args:
+            err: A dictionary with 'index', 'local_id' and 'message' fields,
+                representing the BatchDeleteErrorInfo dictionary that's
+                returned by the server.
+        """
+        self._index = err.get('index', 0)
+        self._local_id = err.get('local_id', "")
+        self._reason = err.get('message', '')
+
+    @property
+    def index(self):
+        return self._index
+
+    @property
+    def reason(self):
+        return self._reason
+
+    @property
+    def local_id(self):
```

Comment:
It seems this type wasn't really mentioned in the API proposal. Should we use `uid` here instead of `local_id`? We haven't exposed `local_id` terminology anywhere else.

Reply:
uid is better than local_id. Done.

> API proposal.

Hm, right. It's listed as ErrorInfo there, which per docs on ErrorInfo, makes it inconvenient to do without a breaking change. But it's *only* the docs; we could instead change them to read "Represents an error encountered while importing an ImportUserRecord *or deleting multiple users*". We'd then also have to eliminate uid/local_id. Which wouldn't be terrible; user's could recover it from the index property combined with their input list.

Reply:
Ok, I've switched to ErrorInfo.